security.html

```diff
@@ -30,3 +30,70 @@ <h3>Key Management</h3>
+    <p>
+        The essensce of the the trust on first use problem, is best highlighted by it's
+        <a target="wikipedia" href="https://en.wikipedia.org/wiki/Trust_on_first_use">Wikipedia article</a>.
+        In summary, the TOFU problem is created because it's not possible to verify with only a public key,
+        who is the owner of the private key. This lack of knowledge gives rise to the introduction of MITM attacks.
+    </p>
+    <p>
+        There's two separate ways to handle the TOFU problem with the peer DID method spec. The more common
+        variant is to bootstrap based upon a trusted third party channel. The new approach which is made
+        possible by Verifiable credentials.
+    </p>
+    <h5>Bootstrap from trusted third party channel</h5>
+    <p>
+        The core premise of using an out of band channel based solution is to be able to bootstrap the new
+        channel of communication from a third party trusted channel of communication. In many cases this is an
+        adequette solution, and in others it may require additional pieces of information in order to establish
+        trust, such as a verifiable credential exchange.
+    </p>
+    <p>
+        This method has been used in previous systems to rely on in person interactions to verify some piece of
+        information. For example, in some end to end messaging apps, they rely on key fingerprinting which
+        require a use to verify that a key fingerprint sent over a trusted communication channel or viewed in
+        person (usually a number) matches between the two parties. This method allows user to verify that the
+        channel hasn't been Man in the Middle (MITM) attacked.
+    </p>
+    <p>
+        There's also ways that the user experience could be improved using this same security process. For
+        example, a <a target="superuser" href="https://superuser.com/questions/22535/
+                what-is-randomart-produced-by-ssh-keygen">randomart image</a> could be embedded into a QR code and then
+        the randomart image could be displayed on the users device as well. The user would then be required to
+        verify the image displayed on the user's device is the same as the one embedded in the QR code. This is
+        a great way to handle short-lived, ephemeral connections such as unlocking a door while being sure the
+        connection has not been MITM attacked.
+    </p>
+    <h5>Binding Verifiable Credentials with Public keys</h5>
+    <p>
+        A verifiable credential method is a new approach to bootstrapping trust entirely within the connection.
+        The idea behind this is that a verifiable credential issued by a third party can be combined with a
+        self-attested attribute in a credential proof to link the possession of the key to a verifiable
+        credential.
+    </p>
+    <p>
+        The reason this works is because the blinded secret acts as a coupling mechanism to prove that the owner
```

Comment:
@kdenhartog instead of "blinded secret", can't we reference concepts already present in https://w3c.github.io/vc-data-model/#zero-knowledge-proofs ?

Even so, support for ZKP in the VC spec seems shaky..

Reply:
The VC spec is written in a way that doesn't allow me to convey why this works. The reason this works is because the blinded link secret is presumed to be owned by one person (the holder of the credential) and can be used to generate a proof that the credential attributes and the self attested attributes were generated by the same entity.

As for the support for ZKP, that's a point that I've seen, but to my understanding is they'll remain in the specification. The reason that was added was because there isn't support in the test suite. See Issue [#662](https://github.com/w3c/vc-data-model/issues/662) where it's mentioned that multiple implementations are being built.

Reply:
@kdenhartog my main concern here is the reference to a very indy-specific technology/concept in an open standardized spec. Moreover, the reference to it is made casually without providing any additional context to the reader who will most likely have no idea what a "blinded secret" is.

Reply:
Gotcha, I think I understand what you're looking for and can make a few edits to the language based on this discussion. What I have in mind is changing:

> The reason this works is because the blinded secret acts as a coupling mechanism to prove that the owner

to

> The reason this works is because the cryptographic proof generated by the holder binds the attributes provided from a verifiable credential with the self attested public key attribute. This is done in a way that only the holder could generate this proof and therefore the recipient can identify all keys recognized by the sender while also being able to identify attributes about the sender in a way that a trust decision can be made.

Reply:
@kdenhartog that would be a more generic way of describing it, yes.

(of course, readers will be left to wonder just how exactly to do this but I suppose this RFC can be later updated to reference a new spec for this)